syntax = "proto3";
package terminus.recording;

import "terminus/proto/core.proto";

message Recording {
  string jwt = 1;
  terminus.core.Typology typology = 2;
  bytes id = 3;
  bytes user_id = 4;
  bytes activity_id = 5;
  string device_id = 6;
  int64 started = 7;
  int64 ended = 8;

  message Point {
    int64 time = 2;
    float speed = 3;
    double position_lat = 4;
    double position_lon = 5;
    float ele = 6;
    float pdop = 7;
    float hdop = 8;  // used for autopause
  }

  repeated Point points = 9;

  message CCU2DataPoint {
    int32 gear = 1;
    int32 rpm = 2;
    int32 throttle = 3;
    int32 mapswitchmode = 4;
  }

  repeated CCU2DataPoint ccu2_data_points = 10;
}
